### PR TITLE
Homepage add components 

### PR DESCRIPTION
**Pull request by sanchay811**
- Added reusable panel components for trending bars on sides
- Improved visual styling
- Google font api included and fontawesome icons library added
- Navbar design changes
- Added a palette of color options to choose from
- Updated logo

### CODE REVIEW

File: src/stylesheet/components/_main.scss [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+.main-section {
+  background: $color1;
+  padding: 10px;
+  border-top: $color2 1px solid;
+  color: $color4;
+}
+
+
+.main-section-header h3 {
```
Comment (howardhenry) [inline src/stylesheet/components/_main.scss]: Nest SASS styles

File: src/stylesheet/components/_navbar.scss [hunk 2]
```diff
@@ -1,7 +1,27 @@
+@import 'custom/palette';
+
 .navbar {
   margin-bottom: 0 !important;
+  background: $color5;
+  border: none;
+  border-radius: 0px;
+}
+
+.navbar > .container > .navbar-collapse > ul > li > a {
```
[the remark this answers is not in the document]
Comment (sanchay811) [reply src/stylesheet/components/_navbar.scss]: 👍

File: src/components/panel/panel.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import React, { Component } from 'react';
+import { Panel } from 'react-bootstrap';
+
+class SimplePanel extends Component {
+  constructor(props) {
```
[the remark this answers is not in the document]
Comment (sanchay811) [reply src/components/panel/panel.js]: Yeah I realized that it was useless to have a constructor here so have removed it and turned into a pure component.
Thanks for the reading materials, quite helpful!

File: src/stylesheet/custom/_palette.scss [hunk 4]
```diff
@@ -0,0 +1,13 @@
+/* HSL */
+$color1: hsla(196%, 93%, 12%, 1);
```
[the remark this answers is not in the document]
Comment (sanchay811) [reply src/stylesheet/custom/_palette.scss]: I was trying to think of better name than custom but had placed that for the time being. Base works much better and clearer as to the purpose.

File: src/stylesheet/custom/_palette.scss [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+$color2: hsla(190%, 64%, 34%, 1);
+$color3: hsla(210%, 78%, 86%, 1);
```
[the remark this answers is not in the document]
Comment (sanchay811) [reply src/stylesheet/custom/_palette.scss]: Agreed, thanks for the feedback. I have incorporated the updates and is much clearer now.

File: .eslintrc [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+    "airbnb"
+  ],
+  "parser": "babel-eslint",
+  "rules": {
+    "arrow-body-style": 0,
+    "arrow-parens": [
+      "error",
+      "always"
+    ],
+    "camelcase": 0,
```
Comment (howardhenry) [inline .eslintrc]: I would set this rule to `"camelcase": 2," as camelcase is the leading best practice convention of defining variables.

File: .eslintrc [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+      "error",
+      2
+    ],
+    "react/jsx-indent-props": [
+      "error",
+      2
+    ],
+    "react/no-array-index-key": 0,
+    "react/no-did-mount-set-state": 0,
+    "react/prefer-stateless-function": 0,
```
Comment (howardhenry) [inline .eslintrc]: You should use stateless functions where possible to improve performance. So, set this to `"react/prefer-stateless-function": 2,`

File: package.json [hunk 8]
```diff
@@ -5,8 +5,10 @@
   "main": "index.js",
   "scripts": {
     "start": "webpack-dev-server --config ./webpack.config.js --mode development",
-    "eslint": "eslint --fix *.js",
-    "test": "echo \"Error: no test specified\" && exit 1"
+    "eslint": "gulp eslint",
+    "sass-lint": "gulp sass-lint",
+    "test": "echo \"Error: no test specified\" && exit 1",
+    "precommit": "npm run eslint && npm run sass-lint"
```
Comment (howardhenry) [inline package.json]: 👍

**Pull request merged**